### ISSUE TITLE
Incomplete handling of Exceptions when reading morph.dat

### ISSUE THREAD

**Issue by radfast** [iChun/Morph#467]
Your method morph.common.core.EventHandler.onWorldLoad() intends to detect corrupt/incomplete morph.dat using the following code, summarised:

try{
Morph.proxy.tickHandlerServer.saveData = CompressedStreamTools.readCompressed(new FileInputStream(file));
}
catch(EOFException e) ... ...

Unfortunately Mojang's code at net.minecraft.nbt.NBTBase.func_130104_b - which is the method which actually reads the bytes from the .dat file - will generally not throw an EOFException when it encounters an End of File.  Almost always, it will instead throw a net.minecraft.util.ReportedException.ReportedException.   (It is relevant to note that NBTBase.func_130104_b is called recursively for a typical NBT data structure which has several layers of depth, due to the call within it to method nbtbase.load().  Therefore, typically, any IOException deep in the NBT data structure will either be caught immediately by the try/catch block in NBTBase.func_130104_b, or will propagate up and be caught by that same try/catch block higher up in the call stack.  This is why Mojang's code almost always throws a net.minecraft.util.ReportedException.ReportedException.  The only situation throwing an EOFException would be an EOF on reading the first byte of what should be a new NBTBase structure, before there are any recursive calls to nbtbase.load().)

TL;DR: For your code to work as you intend it, it needs to catch net.minecraft.util.ReportedException.ReportedException in the same way that it presently catches EOFException.

Special request: when you fix this, please release a new update for the 1.6.4 version of your mod as well, so that all the 1.6.4 modpacks can have it.

I offer no explanation on how the morph.dat is being corrupted in the first place, except to note that all sorts of bad stuff probably happens during world saves on a multiplayer server running Forge, Spigot, Bukkit, Bukkit plugins, and a big modpack.
